src/include/sof/audio/volume.h

```diff
@@ -210,14 +210,16 @@ static inline vol_scale_func vol_get_processing_function(struct comp_dev *dev,
 {
 	struct processing_module *mod = comp_get_drvdata(dev);
 
-	switch (mod->priv.cfg.base_cfg.audio_fmt.depth) {
+	switch (mod->priv.cfg.base_cfg.audio_fmt.valid_bit_depth) {
 	case IPC4_DEPTH_16BIT:
 		return volume_func_map[0].func;
+	case IPC4_DEPTH_24BIT:
+		return volume_func_map[1].func;
```

Comment:
This is a bit conflicting with commit message. The original code could not select 24bit as there was not case for IPC4_DEPTH_24TH, right. But a minor thing.